pkg/oc/cli/admin/node/logs/logs.go

```diff
@@ -0,0 +1,538 @@
+		%[1]s node-logs --role master --path=cron
+	`)
+)
+
+// LogsOptions holds all the necessary options for running oc adm node-logs.
+type LogsOptions struct {
+	Resources []string
+	Selector  string
+	Role      string
+
+	// the log path to fetch
+	Path string
+
+	// --path=journal specific arguments
+	Pattern                string
+	PatternCaseInsensitive bool
+	Units                  []string
+	SinceTime              string
+	UntilTime              string
+	Tail                   int
+	Output                 string
+
+	// output format arguments
+	Raw   bool
+	Unify bool
+
+	RESTClientGetter func(mapping *meta.RESTMapping) (resource.RESTClient, error)
+	Builder          *resource.Builder
+
+	genericclioptions.IOStreams
+}
+
+func NewLogsOptions(streams genericclioptions.IOStreams) *LogsOptions {
+	return &LogsOptions{
+		Path:      "journal",
+		IOStreams: streams,
+	}
+}
+
+// NewCmdLogs creates a new logs command that supports OpenShift resources.
+func NewCmdLogs(baseName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewLogsOptions(streams)
+	cmd := &cobra.Command{
+		Use: "node-logs [-l LABELS] [NODE...]",
+		DisableFlagsInUseLine: true,
+		Short:   "Display and filter node logs",
+		Long:    logsLong,
+		Example: fmt.Sprintf(logsExample, baseName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.RunLogs())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Path, "path", o.Path, "Retrieve the specified path within the node's /var/logs/ folder. The 'journal' value will allow querying the journal on supported operating systems.")
+
+	cmd.Flags().StringSliceVarP(&o.Units, "unit", "u", o.Units, "Return log entries from the specified unit(s). Only applies to node journal logs.")
+	cmd.Flags().StringVarP(&o.Pattern, "search", "S", o.Pattern, "Filter log entries by the provided regex pattern. Only applies to node journal logs.")
+	cmd.Flags().BoolVar(&o.PatternCaseInsensitive, "ignore-case", o.PatternCaseInsensitive, "Make the search filter case-insensitive. Only applies to node journal logs.")
+	cmd.Flags().StringVar(&o.SinceTime, "since", o.SinceTime, "Return logs after a specific ISO timestamp or relative date. Only applies to node journal logs.")
+	cmd.Flags().StringVar(&o.UntilTime, "until", o.UntilTime, "Return logs before a specific ISO timestamp or relative date. Only applies to node journal logs.")
+	cmd.Flags().StringVarP(&o.Output, "output", "o", o.Output, "Display journal logs in an alternate format (short, cat, json, short-unix). Only applies to node journal logs.")
+	cmd.Flags().IntVar(&o.Tail, "tail", o.Tail, "Return up to this many lines from the end of the log. Only applies to node journal logs.")
+
+	cmd.Flags().StringVar(&o.Role, "role", o.Role, "Set a label selector by node role.")
+	cmd.Flags().StringVarP(&o.Selector, "selector", "l", o.Selector, "Selector (label query) to filter on.")
```

Comment:
While we have this option, to select/use a label to direct what node/nodes we want this to run on, is this the best way/experience for a user to get journal logs from a single node or host?

Reply:
Single users would use “node/foo”